Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -540,14 +555,14 @@ public override void BeginCheck(string descriptiveName, VCExpr vc, ErrorHandler
       if (this.libOptions.EmitDebugInformation) {
         SendThisVC("(set-info :boogie-vc-id " + SmtLibNameUtils.QuoteId(descriptiveName) + ")");
       }
-
       if (options.Solver == SolverKind.Z3)
       {
         SendThisVC("(set-option :" + Z3.TimeoutOption + " " + options.TimeLimit + ")");
         SendThisVC("(set-option :" + Z3.RlimitOption + " " + options.ResourceLimit + ")");
         if (options.RandomSeed.HasValue)
         {
-          SendThisVC("(set-option :" + Z3.RandomSeedOption + " " + options.RandomSeed.Value + ")");
+          SendThisVC("(set-option :" + Z3.SmtRandomSeed + " " + options.RandomSeed.Value + ")");
+          SendThisVC("(set-option :" + Z3.SatRandomSeed + " " + options.RandomSeed.Value + ")");
```

Comment:
This change could cause verification regressions as well, right? I can deal with that personally since this isn't a frequently used feature, and we're not committing to zero "regressions" (yet :), but if someone's code is broken by this is there an easy workaround?

Reply:
The workaround would be to find a new random seed that works :-)